shuup/core/models/_products.py

```diff
@@ -568,7 +571,13 @@ def soft_delete(self, user=None):
             # Bypassing local `save()` on purpose.
             super(Product, self).save(update_fields=("deleted",))
 
+    # TODO (2.0): deprecate this
     def verify_mode(self):
+        warnings.warn(
```

Comment:
How about making Product polymorphic and replace the whole concept of ProductMode with product class, i.e. the subclass determines what kind of product it is?  Though maybe that's more like Shuup 3.0 stuff then... :)